src/main/groovy/nextflow/processor/TaskRun.groovy

```diff
@@ -384,7 +385,7 @@ class TaskRun implements Cloneable {
     }
 
     Map<InParam,List<FileHolder>> getInputFiles() {
-        (Map<InParam,List<FileHolder>>) getInputsByType( FileInParam )
+        (Map<InParam,List<FileHolder>>) getInputsByType( FileInParam ) + (Map<InParam,List<FileHolder>>) getInputsByType( FilemapInParam )
     }
```

Comment:
Instead of invoking a second time `getInputsByType` you can specifying `FilemapInParam` class as a second parameter eg. `getInputsByType( FileInParam, FilemapInParam )